src/Storages/StorageDistributed.cpp

```diff
@@ -609,7 +609,7 @@ void StorageDistributed::read(
     ClusterProxy::executeQuery(query_plan, select_stream_factory, log,
         modified_query_ast, local_context, query_info,
         sharding_key_expr, sharding_key_column_name,
-        getCluster());
+        query_info.cluster);
```

Comment:
Deserve a separate commit, since it fixes the race between `getQueryProcessingStage()` and `read()` (if the cluster definition will be changed in between).

Reply:
done